### PR TITLE
Fix arguments type and add typehint.

### PR DESCRIPTION
**Pull request by ADmad**
This also fixes [this](https://travis-ci.org/cakephp/cakephp/jobs/476220922#L713) reported by phpstan at level 3.

### CODE REVIEW

**Comment by ADmad** [comment]
> Do we need to backport to 3.x?

Sure, why not.

New release of psalm is spewing new errors :unamused:.

**Comment by dereuromark** [comment]
yeah looks like a bad release. better to hardcode to previous one.

**Comment by ADmad** [comment]
Nope, they seem like genuine errors which weren't reported before.

File: src/Event/EventManager.php [hunk 1]
```diff
@@ -103,14 +103,14 @@ public function on($eventKey = null, $options = [], $callable = null)
             return $this;
         }
         $argCount = func_num_args();
-        if ($argCount === 2) {
+        if ($eventKey && $argCount === 2) {
```
Comment (saeideng) [inline src/Event/EventManager.php]: `public function on($eventKey = null, `
if `$eventKey` can not be `null` in within this method, and it throw `InvalidArgumentException`
why it is `null`  by default ?

Comment (ADmad) [reply src/Event/EventManager.php]: Good question, I don't see why `$eventKey` is nullable in the interface. I'll remove it.

**Pull request merged**